### ISSUE TITLE
Missing model.train() After evaluate_loss() Call

### ISSUE THREAD

**Issue by minakshisharma197** [minakshisharma197/llm-project#3]
Description:
Inside training loop:

if iter % eval_interval == 0:
    losses = estimate_loss()

estimate_loss() switches the model to .eval() mode and then back to .train(), but if training stops early or user breaks loop, the model may remain in eval mode, causing dropout & layernorm to behave incorrectly.
